File: src/dev/api/NativeAPI.js
var NativeAPIJava = WRAP_JAVA("com.core.api.module.NativeAPI");
var ConversionType = {
    ITEM: 0,
    BLOCK: 1
};
var NativeAPI = {
    dynamicToStatic(id, type){
        return Number(NativeAPIJava.dynamicToStatic(id, type || ConversionType.ITEM));
    },
    staticToDynamic(id, type){
        return Number(NativeAPIJava.staticToDynamic(id, type || ConversionType.ITEM));
    },
    getActorID(ptr){
        return NativeAPIJava.getActorID(ptr);
    },
    getXBlockPos(ptr){
        return Number(NativeAPIJava.getXBlockPos(ptr));
    },
    getYBlockPos(ptr){
        return Number(NativeAPIJava.getYBlockPos(ptr));
    },
    getZBlockPos(ptr){
        return Number(NativeAPIJava.getZBlockPos(ptr));
    },
    getActorById(ptr){
        return NativeAPIJava.getActorById(ptr);
    }
};

File: src/java/Boot/src/com/core/api/module/NativeAPI.java
package com.core.api.module;

public class NativeAPI {
    native public static int dynamicToStatic(int id, int type);
    native public static int staticToDynamic(int id, int type);
    native public static long getActorID(long ptr);
    native public static int getXBlockPos(long ptr);
    native public static int getYBlockPos(long ptr);
    native public static int getZBlockPos(long ptr);
    native public static long getActorById(long id);

    native public static void setWorldGenerator(String json);
}
